elements/pfe-button/demo/pfe-button.html

```diff
@@ -5,175 +5,90 @@
 
 <section>
   <h2 slot="header">Default - Medium size</h2>
-  <pfe-button>
-    <button>Primary</button>
-  </pfe-button>
-  <pfe-button variant="secondary">
-    <button>Secondary</button>
-  </pfe-button>
-  <pfe-button variant="link" icon="link" icon-position="right">
-    <button>Link</button>
-  </pfe-button>
-  <pfe-button variant="link" inline><button>Inline Link</button></pfe-button>
-  <pfe-button variant="link" danger><button>Danger Link</button></pfe-button>
+  <pfe-button>Primary</pfe-button>
+  <pfe-button variant="secondary">Secondary</pfe-button>
+  <pfe-button variant="link" icon="link" icon-position="right">Link</pfe-button>
+  <pfe-button variant="link" inline>Inline Link</pfe-button>
+  <pfe-button variant="link" danger>Danger Link</pfe-button>
 
   <h3>Plain button</h3>
   <pfe-button plain>
-    <button aria-label="Action">
-      <svg fill="currentColor" height="1em" width="1em" viewBox="0 0 352 512">
-        <path d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"></path>
-      </svg>
-    </button>
+    <svg aria-label="Action" fill="currentColor" height="1em" width="1em" viewBox="0 0 352 512">
+      <path d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"></path>
+    </svg>
   </pfe-button>
 
   <h3>Control variant</h3>
-  <pfe-button variant="control"><button>Control</button></pfe-button>
+  <pfe-button variant="control">Control</pfe-button>
   <pfe-button variant="control">
-    <button aria-label="Copy">
-      <svg fill="currentColor" height="1em" width="1em" viewBox="0 0 448 512">
-        <path d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"/>
-      </svg>
-    </button>
+    <svg aria-label="Copy" fill="currentColor" height="1em" width="1em" viewBox="0 0 448 512">
```

Comment:
Use offscreen text and aria-hidden on SVG. See https://www.sarasoueidan.com/blog/accessible-icon-buttons/

Reply:
can / should we handle this inside the web component class, or should we insist that users (and/or pfe-icon) handle it?

Reply:
Ok, how does this strike you (as of b92d36e4):
1. buttons can have slotted text. this is best
2. buttons which are icon-buttons have slotted icons. in this case, the button must have the `label` attribute. (when that attribute is present, the icon container is removed from the AOM and the shadow button get's an aria-label)